src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -430,7 +468,14 @@ private async Task ProcessIncomingAsync(ConnectionContext connection)
 
                             while (ServiceProtocol.TryParseMessage(ref buffer, out var message))
                             {
-                                _ = DispatchMessageAsync(message);
+                                if (message is CloseConnectionMessage closeConnectionMessage && closeConnectionMessage.ConnectionId == Constants.FinAck)
+                                {
+                                    inLoop = false;

```

Comment:
Does this mean the derived type never gets to see the close message? Does that break anything?

Reply:
There is a double check `ConnectionId` == "finack", we only handle `CloseConnectionMessage` with this feature.

Reply:
It's kinda tricky but we don't want to introduce new fields.
What about your opinion? Shall we introduce a new field like `Type`?

Reply:
This is tricky... I'd prefer a new field though...

Reply:
Is it? So introducing a `type` field might seem good in this senario I guess?